app/models/concerns/hud_reports/households.rb

```diff
@@ -115,18 +115,31 @@ def hoh_age(household_id, date)
     end
 
     private def calculate_move_in_date(hh_id, she)
-      return nil unless she.move_in_date.present?
-
       move_in_date = she.move_in_date
       # If the move-in-date is valid, just use it
-      return move_in_date if move_in_date >= she.first_date_in_program
+      return move_in_date if move_in_date.present? && move_in_date >= she.first_date_in_program
 
-      # If the client moved in before the entry date, and the HoH was present on the move-in date, use the
-      # entry date as the move-in date.
+      # Get HoH for further calculations
       household_members = households[hh_id]
       hoh = household_members.detect { |hm| hm[:relationship_to_hoh] == 1 }
-      return nil unless hoh.present?
-      return she.first_date_in_program if hoh[:entry_date] <= move_in_date
+      # HoH does not exist or does not have a move-in date - cannot do further calculations
+      return nil unless hoh.present? && hoh[:move_in_date].present?
+
+      # [Handling Housing Move-In Dates] - https://files.hudexchange.info/resources/documents/HMIS-Standard-Reporting-Terminology-Glossary-2024.pdf
+
+      # Heads of household with [housing move-in dates] prior to their [project start dates] should have the [housing move-in dates] disregarded entirely.
+      return nil unless hoh[:entry_date] <= hoh[:move_in_date]
+
+      # When a household member was already in the household when they became housed (individual’s [project
+      # start date] <= head of household’s [housing move-in date]), the head of household’s [housing move-in date]
+      # should be used as the individual’s [housing move-in date]. If the household member exited before the
+      # household moved into housing, they do not inherit this [housing move-in date].
+      return hoh[:move_in_date] if (she.first_date_in_program..she.exit_date).include?(hoh[:move_in_date])
+
+      # When a household member joins the household after they are already housed (individual’s [project start
+      # date] > head of household’s [housing move-in date]), the individual’s [project start date] should be used as
+      # the individual’s [housing move-in date].
+      return she.first_date_in_program if she.first_date_in_program > hoh[:move_in_date]
```

Comment:
Thanks @dtgreiner this looks great.  One tiny tweak since you have `she.exit_date` let's go ahead and use the aliases for both dates:
```suggestion
      return hoh[:move_in_date] if (she.entry_date..she.exit_date).include?(hoh[:move_in_date])

      # When a household member joins the household after they are already housed (individual’s [project start
      # date] > head of household’s [housing move-in date]), the individual’s [project start date] should be used as
      # the individual’s [housing move-in date].
      return she.entry_date if she.entry_date > hoh[:move_in_date]
```

Reply:
Adding this in and will have the commit pushed to the PR momentarily